pkg/resources/api_authentication_integration_with_authorization_code_grant.go

```diff
@@ -0,0 +1,523 @@
+				showMapping{"enabled", "enabled", integration.Enabled, integration.Enabled, nil},
+			); err != nil {
+				return diag.FromErr(err)
+			}
+
+			enabled, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "ENABLED" })
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthAccessTokenValidity, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_ACCESS_TOKEN_VALIDITY"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthRefreshTokenValidity, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_REFRESH_TOKEN_VALIDITY"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthClientId, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_CLIENT_ID" })
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthClientAuthMethod, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_CLIENT_AUTH_METHOD"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthAuthorizationEndpoint, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool {
+				return property.Name == "OAUTH_AUTHORIZATION_ENDPOINT"
+			})
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthTokenEndpoint, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_TOKEN_ENDPOINT" })
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthAllowedScopes, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_ALLOWED_SCOPES" })
+			if err != nil {
+				return diag.FromErr(err)
+			}
+
+			oauthGrant, err := collections.FindOne(properties, func(property sdk.SecurityIntegrationProperty) bool { return property.Name == "OAUTH_GRANT" })
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			oauthAccessTokenValidityInt, err := strconv.Atoi(oauthAccessTokenValidity.Value)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			oauthRefreshTokenValidityInt, err := strconv.Atoi(oauthRefreshTokenValidity.Value)
+			if err != nil {
+				return diag.FromErr(err)
+			}
+			if err = handleExternalChangesToObjectInDescribe(d,
+				describeMapping{"enabled", "enabled", enabled.Value, enabled.Value, nil},
+				describeMapping{"oauth_access_token_validity", "oauth_access_token_validity", oauthAccessTokenValidityInt, oauthAccessTokenValidityInt, stringToIntNormalizer},
+				describeMapping{"oauth_refresh_token_validity", "oauth_refresh_token_validity", oauthRefreshTokenValidityInt, oauthRefreshTokenValidityInt, stringToIntNormalizer},
```

Comment:
I don't think these normalizers are needed here, usually we just adjust the valueToCompare (check warehouse)

Reply:
Now I'm confused when we should use a normalizer vs normalize `valueToCompare`